src/eva/core/callbacks/writers/embeddings/base.py

```diff
@@ -163,6 +166,21 @@ def _get_item_metadata(
 
         return item_metadata
 
+    def _check_if_exists(self) -> None:
+        """Checks if the output directory already exists and if it should be overwritten."""
+        exists = (
+            os.path.isdir(self._output_dir) and len(list(Path(self._output_dir).rglob("*.pt"))) > 0
```

Comment:
I'm worried a bit that we list all the `.pt` files because (i) what if they are many (ii) if we in the future allow to store embeddings in the cloud, this check will be challenging - maybe I'm missing something though 🤔

Reply:
That's a good point indeed. Solved this by replacing it with suggestion from [here](https://github.com/kaiko-ai/eva/pull/558#discussion_r1671872450)